source-control/contributing/readme.md

```diff
@@ -0,0 +1,151 @@
+
+Good message structure:
+
+* Separate subject from body with a blank line
+* Limit the subject line to 50 characters
+* Capitalize the subject line
+* Do not end the subject line with a period
+* Use the imperative mood in the subject line (*Fix typo in log* vs. *Fixed typo in log* or *Misc fixes in log code*)
+* Wrap the body at 72 characters
+* Use the body to explain what and why vs. how
+* Reference fixed issues with [closing keywords](https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords)
+
+Example of a well structured git commit message:
+
+```md
+Add code review recipe for Go
+
+- Helps teams automate linting and build verification for go projects.
+- Also gives a list of items to verify for go code reviews.
+
+The PR does not add info about VS Code extensions for go, this will
+be added in issue #124
+
+Closes: #123
+```
+
+You can specify the default git editor, which allows you to write your commit messages using your favorite editor. The following command makes Visual Studio Code your default git editor:
+
+```bash
+git config --global core.editor "code --wait"
+```
+
+References:
+
+* [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+* [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+* [On commit messages](http://who-t.blogspot.com/2009/12/on-commit-messages.html)
+* [Information in commit messages](https://wiki.openstack.org/wiki/GitCommitMessages#Information_in_commit_messages)
+* [Git commit best practices](https://medium.com/@nawarpianist/git-commit-best-practices-dab8d722de99)
+
+## Contributing to an existing repository
+
+When working on an existing project, `git clone` the repository and ensure you understand the team's branch, merge and release strategy (e.g. through the projects [CONTRIBUTING.md file](https://blog.github.com/2012-09-17-contributing-guidelines/)).
+
+## Creating a new repository
+
+If you create a new repository, agree on your branch, merge and release strategies upfront and document them in your [CONTRIBUTING.md](https://blog.github.com/2012-09-17-contributing-guidelines/). Also lock the default branches and define who can approve and merge Pull Requests into your default branches.
+
+### Required files in default branch for public repositories
+
+A public repository needs to have the following files in the root directory of the default branch:
+
+* a [LICENSE](Templates/LICENSE) file
+* a [README.md](Templates/README.md) file
+* a [CONTRIBUTING.md](Templates/CONTRIBUTING.md) file
+* reference the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/)
+
+<!-- TODO: the template links to be fixed -->
+
+To start to contribute by creating your own branch, or fork the repository, and commit often - make your commit comments useful to others by including the *WHAT* and *WHY* (instead of the *HOW*).
+
+## Merge strategies
```

Comment:
Note to other reviewers as it is hard to spot this change... this was changed from **Commit history** to **Merge Strategies**
I'm good with either

Reply:
Happy to change, but prefer "merge strategies" as it better describes the topic than "commit history".